src/components/CurrentPathwayProvider.tsx

```diff
@@ -71,6 +74,33 @@ export const CurrentPathwayProvider: FC<CurrentPathwayProviderProps> = memo(({ c
     if (pathway) updatePathway(pathway);
   }, [pathway, updatePathway]);
 
+  // Update criteriaSource if criteria change
+  useEffect(() => {
+    if (!pathway) return;
+
+    const criteriaIds = criteria.map(crit => crit.id);
+    const newPathway = produce(pathway, draftPathway => {
+      Object.entries(draftPathway.nodes).forEach(([nodeIndex, node]) => {
+        node.transitions.forEach(({ condition }, transitionIndex) => {
+          // If a matching criteria does not already exist, try and find one
+          if (condition && !criteriaIds.includes(condition.criteriaSource as string)) {
+            const [library, statement] = condition.cql.split('.');
+            const criteriaSource = criteria.find(
+              crit => crit.elm?.library.identifier.id === library && crit.statement === statement
+            )?.id;
+            // Only update if a criteria source is actually found.
```

Comment:
If no criteria source is actually found, will `immer.produce` still create a new object that is the same?

Previously the call to `setPathways` (`resetCurrentPathways` now in this PR) gets called even if there are no changes.  Does it make sense to have the the `resetCurrentPathway` and `produce` calls nested under the `if (criteriaSource)` so it only updates when a criteria source is available?

Reply:
Good catch - I think it does. The produce had to be brought out though because otherwise some weird recursion happened which crashed the app. I'll add a flag to only call `resetCurrentPathway` if a change happened.